File: sorting_&_searching/go_/quick_sort.go
package main

func LeftRightValue(arr []int) []int {

}

func QuickSort(arr []int) []int {
	if len(arr) == 1 {
		return arr
	}
	pivot := arr[0]
	var LeftArray, RightArray []int
	for _, val := range arr {
		if val <= pivot {
			LeftArray = append(LeftArray, val)
		} else {
			RightArray = append(RightArray, val)
		}
	}
	return QuickSort(LeftArray)
}

File: sorting_&_searching/quick_sort.py
def quick_sort(arr):
    if len(arr) == 1:
        return arr
    pivot = arr[0]
    left = [x for x in arr[1:] if x <= pivot]
    right = [x for x in arr[1:] if x > pivot]
    return quick_sort(left) + [pivot] + quick_sort(right)


array = [5, 4, 2, 1, 4]
print(quick_sort(array))
